src/coreclr/jit/compiler.h

```diff
@@ -8950,22 +8950,34 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         Memcpy  // Copying memory from src to dst
     };
 
-    unsigned int getUnrollThreshold(UnrollKind type)
+    unsigned int getUnrollThreshold(UnrollKind type, bool canUseSimd = true)
     {
         unsigned threshold = TARGET_POINTER_SIZE;
 
 #if defined(FEATURE_SIMD)
-        threshold = maxSIMDStructBytes();
+        if (canUseSimd)
+        {
+            threshold = maxSIMDStructBytes();
 #if defined(TARGET_ARM64)
-        // ldp/stp instructions can load/store two 16-byte vectors at once, e.g.:
-        //
-        //   ldp q0, q1, [x1]
-        //   stp q0, q1, [x0]
-        //
-        threshold *= 2;
+            // ldp/stp instructions can load/store two 16-byte vectors at once, e.g.:
+            //
+            //   ldp q0, q1, [x1]
+            //   stp q0, q1, [x0]
+            //
+            threshold *= 2;
 #elif defined(TARGET_XARCH)
-        // Ignore AVX-512 for now
-        threshold = max(threshold, YMM_REGSIZE_BYTES);
+            // Ignore AVX-512 for now
```

Comment:
can you make this a `TODO-XARCH-AVX512` so we can track it alongside any other AVX-512 cleanup/considerations after more support is added?

Reply:
Sure, added